examples/step-6/doc/results.dox

```diff
@@ -483,3 +483,102 @@ want, is a complex topic in itself. You can find much more on this in
 step-49, step-53, and step-54, among other tutorial programs that cover
 the issue. Information on curved domains can also be found in the
 documentation module on @ref manifold "Manifold descriptions".
+
+
+
+<h4>Playing with the regularity of the solution</h4>
+
+From a mathematical perspective, solutions of the Laplace equation
+@f[
+  -\Delta u = f
+@f]
+on smoothly bounded, convex domains are known to be smooth themselves. The exact degree
+of smoothness, i.e., the function space in which the solution lives, depends
+on how smooth exactly the boundary of the domain is, and how smooth the right
+hand side is. Some regularity of the solution may be lost at the boundary, but
+one generally has that the solution is twice more differentiable in
+compact subsets of the domain than the right hand side.
+If, in particular, the right hand side satisfies $f\in C^\infty(\Omega)$, then   
+$u \in C^\infty(\Omega_i)$ where $\Omega_i$ is any compact subset of $\Omega$
+($\Omega$ is an open domain, so a compact subset needs to keep a positive distance
+from $\partial\Omega$).
+
+The situation we chose for the current example is different, however: we look
+at an equation with a non-constant coefficient $a(x)$:
+@f[
+  -\nabla \cdot (a \nabla u) = f.
+@f]
+Here, if $a$ is not smooth, then the solution will not be smooth either, 
+regardless of $f$. In particular, we expect that wherever $a$ is discontinuous
+along a line (or along a plane in 3d),
+the solution will have a kink. This is easy to see: if, for example, $f$
+is continuous, then intuitively, $\nabla \cdot (a \nabla u)$ needs to be 
+continuous, i.e., $a \nabla u$ must be continuously differentiable 
+(not have a kink). Consequently, if $a$ has a discontinuity, then $\nabla u$
+must have an opposite discontinuity so that the two exactly cancel and their
+product yields a function without a discontinuity. But for $\nabla u$ to have
+a discontinuity, $u$ must have a kink. This is of course exactly what is
+happening in the current example, and easy to observe in the pictures of the
+solution.
+
+In general, if the coefficient $a(x)$ is discontinuous along a line in 2d,
```

Comment:
$a(\mathbf{x})$

BTW. If the position vector is consistently denoted without bold font, then I understand that you'd be reticent to change it here (and above).